test/python/test_roles_api.py

```diff
@@ -4,24 +4,714 @@
+            'group',
+            'userGroup',
+            members='',
+            member=role_id
+        )
+        return response
 
-    def test_get_role(self):
-        """Test case for get_role
+    # Test cases for GET requests to /roles/{account}/{kind}/{identifier} endpoint
 
-        Get role information
+    def test_get_role_200(self):
+        """Test case for 200 status response when getting role data
+        Gets information on the specified role
         """
-        details = self.api.get_role(self.account, 'user', 'admin')
+        response = self.api.get_role_with_http_info(self.account, 'user', 'admin')
+        role_details = response[0]
 
-        self.assertIsInstance(details, dict)
+        self.assertEqual(response[1], 200)
+        self.assertIsInstance(role_details, dict)
         members = ['created_at', 'id', 'members']
         for i in members:
-            self.assertIn(i, details)
+            self.assertIn(i, role_details)
+
+    def test_get_role_400(self):
+        """Test case for 400 status repsonse when getting role data
+        Error originates from NGINX, occurs when making HTTPS requests to Conjur through NGINX
+        This same request made directly to Conjur with HTTP results in a 422 status response
+        400 - request rejected by NGINX proxy
+        """
+        with self.assertRaises(openapi_client.ApiException) as context:
+            self.api.get_role_with_http_info(self.account, NULL_BYTE, 'admin')
+
+        self.assertEqual(context.exception.status, 400)
+
+    def test_get_role_401(self):
+        """Test case for 401 status response when getting role data
+        401 - unauthorized request
+        """
+        with self.assertRaises(openapi_client.ApiException) as context:
+            self.bad_auth_api.get_role_with_http_info(self.account, 'user', 'admin')
+
+        self.assertEqual(context.exception.status, 401)
+
+    def test_get_role_404(self):
+        """Test case for 404 status response when getting role data
+        404 - the role could not be found
+        """
+        with self.assertRaises(openapi_client.ApiException) as context:
+            self.api.get_role_with_http_info(self.account, 'user', 'fakeUser')
+
+        self.assertEqual(context.exception.status, 404)
+
+    def test_get_role_422(self):
+        """Test case for 422 status response when getting role data
+        422 - Conjur recieved a malformed request parameter
+        """
+        with self.assertRaises(openapi_client.ApiException) as context:
+            self.api.get_role_with_http_info(
+                self.account,
+                'user',
+                'admin',
+                members='',
+                search=NULL_BYTE
+            )
+
+        self.assertEqual(context.exception.status, 422)
+
+    # Test cases for getting all of a role's memberships, expanded recursively
+
+    def test_get_all_memberships_200(self):
+        """Test using all query parameter on GET requests
```

Comment:
This comment doesn't fit the format of the rest of your docstrings. In general I don't really care about sticking to a strict format, however I think you should at least throw the return code because it can make it easier to read the `nose2` output.